Consistent formatting of jsdocs comments

trych opened the pull request:
This makes the formatting of all jsdocs inline documentation consistent and therefore cleans it up considerably. After quite a bit of research and testing around, I came up with a formatting style that looks clean, is maintainable and works with the doc generation (once we have the markdown rendering in place).

Here is a before/after of the `margins()` function

Before
```js
/**
 * Sets the margins of a given page. If 1 value is given, all 4 sides are set equally. If 4 values are given, the current page will be adjusted. Adding a 5th value will set the margin of a given page. Calling the function without any values, will return the margins for the current page.
 *
 * @cat Document
 * @subcat Page
 * @method margins
 * @param {Number} [top] Top margin or all if only one.
 * @param {Number} [right] Right margin.
 * @param {Number} [bottom] Bottom margin.
 * @param {Number} [left] Left margin.
 * @param {Number} [pageNumber] Sets margins to selected page, currentPage() if left blank.
 * @return {Object} Current page margins with the properties: top, right, bottom, left.
 */
```

After
```js
/**
 * @description Sets the margins of a given page. If 1 value is given, all 4
 *          sides are set equally. If 4 values are given, the current page will
 *          be adjusted. Adding a 5th value will set the margin of a given page.
 *          Calling the function without any values, will return the margins for
 *          the current page.
 *
 * @cat     Document
 * @subcat  Page
 * @method  margins
 *
 * @param   {Number} [top] Top margin or all if only one.
 * @param   {Number} [right] Right margin.
 * @param   {Number} [bottom] Bottom margin.
 * @param   {Number} [left] Left margin.
 * @param   {Number} [pageNumber] Sets margins to selected page, currentPage()
 *          if left blank.
 * @return  {Object} Current page margins with the properties: `top`, `right`,
 *          `bottom`, `left`.
 */
```

As you can see, this now also strictly stays within the 80-characters line limit to be better readable in most editors.

The syntax looks like this now:
```js
/**
 * @description The description of the method. If the description is longer, it
 *          wraps around to an indentation value that aligns with other
 *          information below.
 *
 * @cat     Category
 * @subcat  Sub-Category
 * @method  methodName
 *
 * @param   {Type} name A parameter.
 * @param   {Type} name A parameter with a description that is long enough that
 *          it wraps into the next line.
 * @param   {Type} [name] An optional parameter.
 * @return  {Type} Description of the return value. This also wraps into the
 *          next line if the description is long enough.
 *          
 * @example <caption>How to use this function</caption>
 * var foo = methodName(); // code lines must not use indentation
 */
```

All methods have been changed to adhere to this new syntax. Markdown was added for inline code snippets and for lists.

trych commented:
@basiljs Does anyone have any comments on this? Else I'll just go ahead and merge it.

fabianmoronzirfas commented:
Great that you cleaned that up. I don't like manual indent of the description. It could be interpreted as Markdown code blocks if we parse the description.  

+1000 for the cleanup
-- on the manual indent

trych commented:
It's not not interpreted as markdown code blocks during parsing, I tested this quite intensely. It parses just fine.

trych commented:
(The scenario you describe would only happen, if there were some blank lines in between blocks, which is never the case)

trych commented:
There is of course a way to [create and update this indentation automatically](https://github.com/20Tauri/DoxyDoxygen) in Sublime.  I could do a write up in the Wiki, about how to set up the formatting.

Would that be okay for you?

trych commented:
Also, the reason I went with indentation in the first place is, because I did some research about what the standard is of how to format such jsdoc matter and found this thread: [What is the proper/canonical formatting of long JSDoc lines?](https://stackoverflow.com/questions/30676286/what-is-the-proper-canonical-formatting-of-long-jsdoc-lines)

The accepted answer there points out that even Google recommends in its JavaScript styleguides to indent lines after the first one. So I thought this seems very reasonable.

fabianmoronzirfas commented:
>There is of course a way to create and update this indentation automatically in Sublime. I could do a write up in the Wiki, about how to set up the package.

>Would that be okay for you

I'm actually not using SublimeText anymore. 

I guess it is not about making Fabian happy. @basiljs had anyone else an opinion or is it just me complaining?

fabianmoronzirfas commented:
FYI 
![8483151c-6add-4a0d-ae50-a3485ecd2f53](https://user-images.githubusercontent.com/315106/38768567-38e1f336-3ff6-11e8-8a5b-dca9e523e11f.png)

fabianmoronzirfas commented:
The display depends very much on the size of the window.

trych commented:
> I'm actually not using SublimeText anymore.

But how do you run ExtendScript? :D

> The display depends very much on the size of the window.

Yes, but that it does not render well on mobile phones is not really an argument, I think. If it were about that, we should also do without the 80-character line limit and it would look much better. But nobody is editing basil.js source code on their mobile? Or are they?

I too would like to hear the opinions of the others. @basiljs ?

And @fabianmoronzirfas, what alternative do you actually prefer? A or B?

A)

```js
/**
 * @description Sets the margins of a given page. If 1 value is given, all 4
 * sides are set equally. If 4 values are given, the current page will be
 * adjusted. Adding a 5th value will set the margin of a given page. Calling the
 * function without any values, will return the margins for the current page.
 *
 * @cat     Document
 * @subcat  Canvas
 * @method  margins
 *
 * @param   {Number} [top] Top margin or all if only one.
 * @param   {Number} [right] Right margin.
 * @param   {Number} [bottom] Bottom margin. Longer descriptions will of course
 *          wrap into the next line again. This happens quite a bit in the
 *          basil.js code base. 
 * @param   {Number} [left] Left margin.
 * @param   {Number} [pageNumber] Sets margins to selected page, currentPage()
 *          if left blank.
 * @return  {Object} Current page margins with the properties: `top`, `right`,
 *          `bottom`, `left`.
 */
```

B
```js
/**
 * @description Sets the margins of a given page. If 1 value is given, all 4
 * sides are set equally. If 4 values are given, the current page will be
 * adjusted. Adding a 5th value will set the margin of a given page. Calling the
 * function without any values, will return the margins for the current page.
 *
 * @cat     Document
 * @subcat  Canvas
 * @method  margins
 *
 * @param   {Number} [top] Top margin or all if only one.
 * @param   {Number} [right] Right margin.
 * @param   {Number} [bottom] Bottom margin. Longer descriptions will of course
 * wrap into the next line again. This happens quite a bit in the basil.js code
 * base. 
 * @param   {Number} [left] Left margin.
 * @param   {Number} [pageNumber] Sets margins to selected page, currentPage()
 * if left blank.
 * @return  {Object} Current page margins with the properties: `top`, `right`,
 * `bottom`, `left`.
 */
```

trych commented:
Wait, what?

[Here](https://github.com/basiljs/basil.js/issues/266#issuecomment-376915880) you specifically stated you were undecided about having description text all in one line because you said, and I quote

> Might reduce readability when user has no line-wrap enabled

So... you changed your mind now?

trych commented:
I mean have it all in one line I find easier to maintain, but it renders awefully on GitHub. And others (p5.js, Processing) also seem to avoid this ...

fabianmoronzirfas commented:
Well. Yes. Seeing the text broken up in favour of line length makes me think that this does not make sense. But we don't need to fight about it. I just think having to indent the text of a description will not happen the time. We will have comments in different styles in the future.

trych commented:
> We will have comments in different styles in the future.

What do you mean by that? That jsdocs/documentation.js is replaced at some point? Or that the 80-character limit is slowly fading away?

trych commented:
Yes, but then it's easy to catch in a PR review. And also, someone will forget to style the comments either way, no matter how we decide. But okay, I also don't want to fight, yet I still want have a decision on what we should do.

I myself am a bit torn, I now put hours of work into getting it to look like this in the whole thing, but I can also understand the appeal of having no line breaks at all.

Any opinions from the others? @basiljs
